.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,28 @@
+name: Lint Go Code
+
+on:
+  pull_request:
+    paths:
+      - '**/*.go'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.19.0
+
+    - name: Install golangci-lint
+      run: go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+
+    - name: Run golangci-lint
+      run: golangci-lint run --allow-serial-runners
+
+    - name: Fix Go code with golangci-lint (optional)
+      run: golangci-lint run --fix --allow-serial-runners
```

Comment:
I think this step will not run if the previous step failed.
If CI step is making changes to files (like lint fix), they have to be committed and pushed to take effect, otherwise, they just go down with the CI container.

My suggestion is to just verify in CI that `make lint` is successful, and if not, the user will have to run locally `make golangci-lint-fix` and push a fixed version  after linting